//
// PROJECT:         Aspia Remote Desktop
// FILE:            proto/file_transfer_session.proto
// LICENSE:         Mozilla Public License Version 2.0
// PROGRAMMERS:     Dmitry Chapyshev (dmitry@aspia.ru)
//

syntax = "proto3";

option optimize_for = LITE_RUNTIME;

import "file_transfer_session_message.proto";

package aspia.proto.file_transfer;

message HostToClient
{
    RequestType type             = 1;
    RequestStatus status         = 2;
    DriveList drive_list         = 3;
    FileList file_list           = 4;
    DirectorySize directory_size = 5;
    FilePacket file_packet       = 6;
}

message ClientToHost
{
    RequestType type                                = 1;
    FileListRequest file_list_request               = 2;
    DirectorySizeRequest directory_size_request     = 3;
    CreateDirectoryRequest create_directory_request = 4;
    RenameRequest rename_request                    = 5;
    RemoveRequest remove_request                    = 6;
    FilePacketRequest file_request                  = 7;
    FilePacket file_packet                          = 8;
}
